cmake/modules/extensions.cmake

```diff
@@ -82,6 +82,7 @@ function(zephyr_sources)
       message(FATAL_ERROR "zephyr_sources() was called on a directory")
     endif()
     target_sources(zephyr PRIVATE ${arg})
+    generate_unique_source_file_ids(${arg})
```

Comment:
please note, you are now significantly changing this command from purely wrapping vanilla CMake to actually do more than what is described:
https://github.com/zephyrproject-rtos/zephyr/blob/f3017aa787e75c77bb7deb9f4ae45d4b4e93fbc3/cmake/modules/extensions.cmake#L47-L57

i'm really not convinced we should do this.

Reply:
Secondly, this will break as soon as someone tries to use a generator expression for adding Zephyr sources, for example like this:
https://github.com/zephyrproject-rtos/zephyr/blob/f3017aa787e75c77bb7deb9f4ae45d4b4e93fbc3/samples/tfm_integration/tfm_secure_partition/dummy_partition/CMakeLists.txt#L28-L29